lib/collection/src/shards/local_shard/clock_map.rs

```diff
@@ -35,8 +38,16 @@ impl ClockMap {
         Ok(clock_map)
     }
 
-    pub fn store(&self, path: &Path) -> Result<()> {
+    pub fn store(&mut self, path: &Path) -> Result<()> {
```

Comment:
maybe AtomicBool instead?

Reply:
We only store from mutable context. It also prevents us from calling it multiple times concurrently.

I'd suggest to change to an atomic boolean if we ever need this from immutable context.